client/lib/signup/step-actions/index.js

```diff
@@ -674,15 +693,8 @@ export function addPlanToCart( callback, dependencies, stepProvidedItems, reduxS
 		return;
 	}
 
```

Comment:
Removing this plugin addition in this `step-action` could potentially affect all usages of this method in all the steps found in `steps-pure.js` module. **However**, I have checked all of the definitions ([example given here](https://github.com/Automattic/wp-calypso/blob/ec0e3184ab999c86718c988ef3c4ddd86240f649/client/signup/config/steps-pure.js#L87)), and none use either `plugin` or `billing_period` in the array of `dependencies,` so none can expect that dependency to exist. 

Additionally, I have checked all the `providesDependenciesInQuery` for the flows and there is no additional occurrence of that apart from the [expected one](https://github.com/Automattic/wp-calypso/blob/ec0e3184ab999c86718c988ef3c4ddd86240f649/client/signup/config/flows-pure.js#L165) so removing `pluginItem` code looks safe here ✅ .

Reply:
That's right, this part was added on the `with-plugin` flow addition, and now I searched and saw it was only being used for this, so I found it safe to remove. Thanks for checking all of it on the review 🙏